lib/pure/os.nim

```diff
@@ -1688,13 +1688,28 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo =
     rawToFormalFileInfo(rawInfo, path, result)
 
 proc isHidden*(path: string): bool =
-  ## Determines whether a given path is hidden or not. Returns false if the
-  ## file doesn't exist. The given path must be accessible from the current
-  ## working directory of the program.
+  ## Determines whether the given ``path`` is hidden or not, using this reference https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory
   ##
-  ## On Windows, a file is hidden if the file's 'hidden' attribute is set.
-  ## On Unix-like systems, a file is hidden if it starts with a '.' (period)
-  ## and is not *just* '.' or '..' ' ."
+  ## On Windows: a path is hidden if it exists and it's "hidden" attribute is set.
+  ##
+  ## On posix: a hidden path starts with ``.`` (see examples).
+  ##
+  ## On OSX, in addition to the "dotfile" behaviour, existing paths with the
+  ## "Invisible" attribute are hidden in Finder, although not in ``ls``.
+  ## Note: the behavior of certain corner cases (see examples) is
+  ## unspecified, subject to change, but can be avoided using ``ospaths.normalizePath``.
+  runnableExamples:
+    when defined(posix):
+      doAssert ".foo".isHidden
+      doAssert(not ".foo/bar".isHidden)
+      doAssert(not ".".isHidden)
+      doAssert(not "..".isHidden)
+      doAssert(not "".isHidden)
+      # TODO: decide on these:
```

Comment:
Why are these subject to change? Let's decide on them now: the best solution is to be consistent with other languages.

Reply:
@dom96 

* unfortunately "other languages" just means java (for the other languages, all i could find was answers on stackoverflow)
* here's what i get for java (java.io.File.isHidden):
```
.foo isHidden: true
.foo/ isHidden: true # because trailing `/` is irrelevant
.foo// isHidden: true
.foo/. isHidden: true # because "." considered hidden
.foo//. isHidden: true
.foo/.. isHidden: true # because ".." is considered hidden
foo isHidden: false
foo/ isHidden: false
foo// isHidden: false
foo/. isHidden: true # because "." considered hidden
foo//. isHidden: true
foo/.. isHidden: true
.. isHidden: true
. isHidden: true
 isHidden: false # LHS is empty => baseName is empty, therefore not starting with "."
/ isHidden: false
```

from which I gather that algorithm for java:
* return true iff baseName(path) starts with "." (whether . or .. or .baz)
where C's convention is used for `baseName` instead of Nim's convention (see https://github.com/nim-lang/Nim/issues/8341)

I actually like java's behavior in all cases above and I think we should follow it.

So my recommendation is:
* accept and implement C's `baseName` https://github.com/nim-lang/Nim/issues/8341 (I can do it if you guys agree on that semantics)
* follow java behavior for `isHidden`

sounds good?

Reply:
So as I've said in #8341 I think you should use ``extractFilename``.

Reply:
resolved all "subject to change" items, and used the new `lastPathPart` proc instead of `extractFileName` so that `isHidden(".git/")` returns true, as it should.